pkg/vm/engine/tae/moengine/relation.go

```diff
@@ -132,8 +136,36 @@ func (rel *txnRelation) Write(_ uint64, bat *batch.Batch, _ engine.Snapshot) err
 	return rel.handle.Append(bat)
 }
 
-func (rel *txnRelation) Delete(_ uint64, _ *vector.Vector, _ string, _ engine.Snapshot) error {
-	panic(any("implement me"))
+func (rel *txnRelation) Delete(_ uint64, data *vector.Vector, col string, _ engine.Snapshot) error {
+	schema := rel.handle.GetMeta().(*catalog.TableEntry).GetSchema()
+	logutil.Debugf("Delete col: %v", col)
+	if schema.HasPK() {
+		if len(schema.SortKey.Defs) > 1 {
```

Comment:
No need to panic here. Schema is only readonly here

Reply:
OK!I have removed it.